crates/ra_assists/src/handlers/unwrap_block.rs

```diff
@@ -0,0 +1,435 @@
+use crate::{Assist, AssistCtx, AssistId};
+
+use ast::LoopBodyOwner;
+use ra_fmt::unwrap_trivial_block;
+use ra_syntax::{ast, AstNode};
+
+// Assist: unwrap_block
+//
+// This assist removes if...else, for, while and loop control statements to just keep the body.
+//
+// ```
+// fn foo() {
+//     if true {<|>
+//         println!("foo");
+//     }
+// }
+// ```
+// ->
+// ```
+// fn foo() {
+//     println!("foo");
+// }
+// ```
+pub(crate) fn unwrap_block(ctx: AssistCtx) -> Option<Assist> {
+    let res = if let Some(if_expr) = ctx.find_node_at_offset::<ast::IfExpr>() {
+        // if expression
+        let mut expr_to_unwrap: Option<ast::Expr> = None;
+        for block_expr in if_expr.blocks() {
+            if let Some(block) = block_expr.block() {
```

Comment:
Maybe we could add a help function to handle these block checking logic ? Seem like duplicate for each cases.